nomos-services/key-management-system/src/keys/ed25519.rs

```diff
@@ -1,20 +1,49 @@
-use bytes::Bytes;
+use bytes::Bytes as RawBytes;
 use ed25519_dalek::ed25519::signature::Signer as _;
-use overwatch::DynError;
 use serde::{Deserialize, Serialize};
 use zeroize::ZeroizeOnDrop;
 
-use crate::secure_key::SecuredKey;
+use crate::{
+    encodings::Encoding,
+    keys::{secured_key::SecuredKey, KeyError},
+};
+
+type InnerEncoding = crate::encodings::Bytes;
```

Comment:
Im not so sure about all this encoding naming to be honest. I would just use `Bytes` explicitly.